### PR TITLE
OAST Trim database

### PR DESCRIPTION
**Pull request by psiinon**
## Overview
The permanent db will just keep getting bigger - this PR allows users to limit the number of days records are kept, default 45.

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

File: addOns/oast/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 ### Added 
 - API support.
 - Raise alerts for OAST interactions that happened in other sessions.
+- Options to trim the database.
```
Comment (kingthorin) [inline addOns/oast/CHANGELOG.md:12]: Be more specific that this is OAST or Permanent?

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java [hunk 2]
```diff
@@ -68,4 +68,64 @@ public AlertEntity getAlertForPayload(String payload) {
...
+        LOGGER.debug("Trimming records older than {} days", days);
+        try {
+            // Yes, passing parameters in this way is horrible!
+            // But I couldn't get the parameters working the "correct" way and we know its
+            // definitely a positive int..
+            String dateClause = "DATE_SUB(CURRENT_TIMESTAMP, INTERVAL " + days + " DAY)";
+
+            Object res =
+                    runQuery(
+                            "DELETE FROM BOAST WHERE REGISTERED_TIMESTAMP < " + dateClause,
```
Comment (kingthorin) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java:85]: Only boast?

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java [hunk 3]
```diff
@@ -68,4 +68,64 @@ public AlertEntity getAlertForPayload(String payload) {
...
+        } catch (Exception e) {
+            LOGGER.error("Failed to trim db", e);
+        }
+    }
+
+    public void clearAllRecords() {
+        try {
+            Object res = runQuery("DELETE FROM BOAST", null, false);
+            if (Integer.parseInt(res.toString()) > 0) {
+                LOGGER.info("Number of old BOAST records trimmed: {}", res);
```
Comment (kingthorin) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java:115]: Only boast?

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java [hunk 4]
```diff
@@ -68,4 +68,64 @@ public AlertEntity getAlertForPayload(String payload) {
...
+                    runQuery(
+                            "DELETE FROM MESSAGE WHERE CREATETIMESTAMP < " + dateClause,
+                            null,
+                            false);
+            if (Integer.parseInt(res.toString()) > 0) {
+                LOGGER.info("Number of old MESSAGE records trimmed: {}", res);
+            }
+
+        } catch (Exception e) {
+            LOGGER.error("Failed to trim db", e);
```
Comment (kingthorin) [inline addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java:107]: Mention OAST or permanent here too?

File: addOns/oast/CHANGELOG.md (hunk 1 above)
Comment (psiinon) [reply addOns/oast/CHANGELOG.md:12]: Well, its in the OAST add-on, but yeah, its trimming the OWASP perm db records

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java (hunk 2 above)
Comment (psiinon) [reply addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java:85]: yes, we handle each table separately.

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java (hunk 3 above)
Comment (psiinon) [reply addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java:115]: as above :)

File: addOns/oast/CHANGELOG.md (hunk 1 above)
Comment (psiinon) [reply addOns/oast/CHANGELOG.md]: Updated

File: addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java (hunk 4 above)
Comment (psiinon) [reply addOns/oast/src/main/java/org/zaproxy/addon/oast/internal/OastPermanentDatabase.java]: Updated

**Review by kingthorin — APPROVED** [review]

**Pull request merged**